go-controller/pkg/clustermanager/userdefinednetwork/controller.go

```diff
@@ -142,6 +143,48 @@ func (c *Controller) Run() error {
 	return nil
 }
 
+// initializeNamespaceTracker populates the namespace-tracker with NAD namespaces who owned by the controller.
+func (c *Controller) initializeNamespaceTracker() error {
+	cudns, err := c.cudnLister.List(labels.Everything())
+	if err != nil {
+		return err
+	}
+	if len(cudns) == 0 {
+		return nil
+	}
+
+	nads, err := c.nadLister.List(labels.Everything())
+	if err != nil {
+		return err
+	}
+	if len(nads) == 0 {
+		return nil
+	}
+	indexedNADs := map[string]netv1.NetworkAttachmentDefinition{}
+	for _, nad := range nads {
+		if nad != nil {
+			indexedNADs[nad.Namespace+"/"+nad.Name] = *nad.DeepCopy()
+		}
+	}
+
+	for _, cudn := range cudns {
+		cudnCopy := cudn.DeepCopy()
```

Comment:
I don't think you need to deepcopy here, as we are not changing cudns in this function?

Reply:
Done